edge/pkg/edged/edged.go

```diff
@@ -873,6 +873,41 @@ func (e *edged) syncLoopIteration(plegCh <-chan *pleg.PodLifecycleEvent, houseke
 				}
 				e.podAdditionQueue.Add(key.String())
 			}
+
+		case update := <-e.readinessManager.Updates():
+			ready := update.Result == proberesults.Success
+			e.statusManager.SetContainerReadiness(update.PodUID, update.ContainerID, ready)
+			pod, ok := e.podManager.GetPodByUID(update.PodUID)
+			if !ok {
+				// If the pod no longer exists, ignore the update.
+				klog.V(4).InfoS("SyncLoop (probe): ignore irrelevant update",
+					"probe", "readiness", "status", map[bool]string{true: "ready", false: ""}[ready], "update", update)
+				break
+			}
+
+			key := types.NamespacedName{
+				Namespace: pod.Namespace,
+				Name:      pod.Name,
+			}
+			e.podAdditionQueue.Add(key.String())
```

Comment:
Is it needed to add the key to `podAdditionQueue`?

Reply:
probe result is changed  with last probe results when we receive probe updates, so this logic is not frequently，and the main reason is that  it can speeds up pod status updates after probe results, see https://github.com/kubernetes/kubernetes/pull/98376#issuecomment-773200720

Reply:
Sounds good.